test/FeeCollector.t.sol

```diff
@@ -0,0 +1,539 @@
+        vm.prank(CONTRACT_DEPLOYER);
+        feeCollector.setClientRate(_clientRate);
+
+        // Post-act data
+        uint256 postClientRate = feeCollector.clientRate();
+
+        // Assertions
+        assertEq(postClientRate, _clientRate);
+    }
+
+    /// @dev
+    // - The clientRate in FeeCollector contract cannot be < 30 or > 100
+    function testFuzz_CannotSetClientRateOutOfRange(uint256 _clientRate) external payable {
+        // Assumptions
+        vm.assume(_clientRate < 30 || _clientRate > 100);
+
+        // Act
+        vm.prank(CONTRACT_DEPLOYER);
+        vm.expectRevert(FeeCollector.OutOfRange.selector);
+        feeCollector.setClientRate(_clientRate);
+    }
+
+    /// @dev
+    // - It should revert with Unauthorized() error when called by an unauthorized sender.
+    function testFuzz_CannotSetClientRateUnauthorized(uint256 _clientRate, address _sender) external payable {
+        // Assumptions
+        _clientRate = bound(_clientRate, 30, 100);
+        vm.assume(_sender != CONTRACT_DEPLOYER);
+
+        // Act
+        vm.prank(_sender);
+        vm.expectRevert(abi.encodeWithSelector(OwnableUnauthorizedAccount.selector, _sender));
+        feeCollector.setClientRate(_clientRate);
+    }
+
+    /// @dev
+    // - The FeeCollector's native balance should decrease by the amount transferred.
+    // - The owner's native balance should increase by the amount transferred.
+    function testFuzz_ExtractNative(uint256 _amount) public {
+        // Setup: fund contract with _amount of native token
+        _amount = bound(_amount, 1, 1e22);
+        vm.deal(feeCollectorAddr, _amount);
+
+        // Get pre-act balances
+        uint256 preContractBalance = feeCollectorAddr.balance;
+        uint256 preOwnerBalance = CONTRACT_DEPLOYER.balance;
+
+        // Assertions
+        assertEq(preContractBalance, _amount);
+
+        // Act
+        vm.prank(CONTRACT_DEPLOYER);
+        feeCollector.extractNative();
+
+        // Ge post-act balances
+        uint256 postContractBalance = feeCollectorAddr.balance;
+        uint256 postOwnerBalance = CONTRACT_DEPLOYER.balance;
+
+        // Assertions
+        assertEq(postContractBalance, 0);
+        assertEq(postOwnerBalance, preOwnerBalance + _amount);
+    }
+
+    /// @dev
+    // - It should revert with Unauthorized() error when called by an unauthorized sender.
+    function testFuzz_CannotExtractNative(uint256 _amount, address _sender) public {
+        // Setup: fund contract with _amount of native token
+        _amount = bound(_amount, 1, 1e22);
+        vm.assume(_sender != CONTRACT_DEPLOYER);
+        vm.deal(feeCollectorAddr, _amount);
+
+        // Act: attempt to extract native
+        vm.prank(_sender);
+        vm.expectRevert(abi.encodeWithSelector(OwnableUnauthorizedAccount.selector, _sender));
+        feeCollector.extractNative();
+    }
+
+    /// @dev
+    // - The FeeCollector's token balance should decrease by its total balance - totalClientBalances
```

Comment:
For the specified feeToken, the FeeCollector's balance should decrease by (total balance - totalClientBalances).